frontend/screens/AlbumScreen.tsx

```diff
@@ -30,28 +33,104 @@ const AlbumScreen = ({route, navigation}) => {
   const [isLoading, setIsLoading] = useState(true);
   const [data, setData] = useState([]);
   const [error, setError] = useState(null);
-
   const [saved, setSaved] = useState(false);
+  const [privatePlaylist, setIsPrivatePlaylist] = useState(false);
 
+  // where we push current copy of songs to the API
   const updateSaved = () => {
     setSaved(true);
-    setModalVisible(true);
+    setIsLoading(true);
+    const NEW_ENDPOINT = `http://localhost:4000/spotify/createSpotifyPlaylist/${authToken}`
+    const NEW_OPTIONS = {
+      method: 'POST',
+      body: JSON.stringify({
+        "name": name,
+        "public": !privatePlaylist, // opposite of private is public :)
+        "trackIds": data.map(song => {return song.id})
+      }),
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }
+    fetch(NEW_ENDPOINT, NEW_OPTIONS)
+      .then(results => {
+        setIsLoading(false);
+        setAddModalVisible(true);
+      })
+      .catch(err => {
+        setIsLoading(false);
+        console.error(err);
```

Comment:
can probably handle errors better after the demo.
the backend has status codes for errors encountered. Maybe for each code, u can show a different message to the user.

Reply:
Hi Hung, I'm not sure where I should be accessing the proper error status code from TheOne? Error 401 in the response body doesn't correspond to a specific response description on Swagger, just seems like it's either 400 or 404 that have descriptions. If I should be referencing error 404, there's no explicit mention of it in the response body either
![image](https://user-images.githubusercontent.com/21225415/147867475-792803a7-82c6-4c3c-bac9-469f1da30de2.png)
![image](https://user-images.githubusercontent.com/21225415/147867499-ada58fc4-d694-45ff-993f-092957a96083.png)